src/business-logic/services/rundown-lock-service.ts

```diff
@@ -0,0 +1,125 @@
+import { Part } from '../../model/entities/part'
+import { Piece } from '../../model/entities/piece'
+import { Owner } from '../../model/enums/owner'
+import { InTransition } from '../../model/value-objects/in-transition'
+import { RundownService } from './interfaces/rundown-service'
+
+const RUNDOWN_LOCK_INTERVAL = 500
+const RUNDOWN_LOCK_ERROR_TEXT = 'Unable to do action. An action was already executed less than 500ms ago'
+
+export class RundownLockService implements RundownService {
+  private static instance: RundownService
+  private hasEnoughTimePassed: boolean = true
+
+  constructor(private readonly rundownService: RundownService) {}
+
+  public static getInstance(rundownService: RundownService): RundownService {
+    if (!this.instance) {
+      this.instance = new RundownLockService(rundownService)
+    }
+    return this.instance
+  }
+
+  public takeNext(rundownId: string): Promise<void> {
+    if (!this.hasEnoughTimePassed) {
+      throw new Error(RUNDOWN_LOCK_ERROR_TEXT)
```

Comment:
We try to avoid to throw `Error`, but instead have more specific errors like `UnsupportedOperationException` or `AlreadyActivatedException`.
Take a look in `/model/exceptions` to see if you can find an exception that fits your need. If not, feel free to create a new one (hint: We need a new for this :) ).

This also gives us the oppertunity to provide an `ErrorCode` to our exception. I can see we already have `ErrorCode.RUNDOWN_IS_LOCKED`, but it has yet to be used for an exception.

Alternatively, you can also just write:
```
throw new Exception(ErrorCode.RUNDOWN_IS_LOCKED, {yourErrorMessage})
```

Reply:
I create new `LockRundownException`